README.md

```diff
@@ -35,6 +35,32 @@ pipeline {
 }
 ```
 
+## Using Optional Parameters
+
+The plugin provides provision to use optional parameters for support of build metadata, pre-release information, settnig the start tag, etc.
+
+### Build Metadata 
+
+`buildMetadata` an optional parameter can be added as follows:
+
+```
+pipeline {
+    agent any
+    environment {
+        NEXT_VERSION = nextVersion(buildMetadata: '001')
```

Comment:
In the issue @garethjevans propose to use `"$env.BUILD_NUMBER"` , I think it's a good idea to use it in the example to understand how is the best usage for this option instead a static value ?

The only things is I don't remember if the `$env.BUILD_NUMBER` is usable in an environment block 🤔  ?

```suggestion
        NEXT_VERSION = nextVersion(buildMetadata: "$env.BUILD_NUMBER")
```

Reply:
Makes sense, I will test this locally to be sure :v: